### PR TITLE
Parser: Redeclaration of gl_CullDistance is disallowed mistakenly.

### PR DESCRIPTION
**Pull request by amdrexu**
I have this fragment shader which causes the compilation fails. Redeclaration of `gl_CullDistance `should be allowed.

```
#version 450

layout(location = 0) out float o1;
layout(location = 1) out float o2;

layout(binding = 0) uniform Uniforms
{
    int i;
};

in float gl_ClipDistance[6];
in float gl_CullDistance[2];

layout(location = 1) in float in1;

void main()
{
    o1 = gl_ClipDistance[3] + gl_CullDistance[1];
    o2 = gl_ClipDistance[i] + gl_CullDistance[i] + in1;
}
```

### CODE REVIEW

File: glslang/MachineIndependent/linkValidate.cpp [hunk 1]
```diff
@@ -390,8 +390,8 @@ void TIntermediate::finalCheck(TInfoSink& infoSink)
     if (invocations == TQualifier::layoutNotSet)
         invocations = 1;
 
-    if (inIoAccessed("gl_ClipDistance") && inIoAccessed("gl_ClipVertex"))
-        error(infoSink, "Can only use one of gl_ClipDistance or gl_ClipVertex (gl_ClipDistance is preferred)");
+    if ((inIoAccessed("gl_ClipDistance") || inIoAccessed("gl_CullDistance")) && inIoAccessed("gl_ClipVertex"))
+        error(infoSink, "Can only use one of gl_ClipDistance (gl_CullDistance) or gl_ClipVertex (gl_ClipDistance is preferred)");
```
Comment (johnkslang) [inline glslang/MachineIndependent/linkValidate.cpp]: This needs to be more clear.  I suspect leaving the existing test as it was, and just adding more tests, with simpler error messages will work better.

**Comment by johnkslang** [comment]
This change looks in the right direction, but with this differ, it will be much easier to tell without the line number changes.  It's also failing the bot tests, but I don't know why.

Can you try again, but with two commits, one that just adds blank lines to the test to cause a line-number change only, and then one that adds the redeclaration without changing line numbers, to just show the functional difference?

**Comment by antiagainst** [comment]
For the buildbot failure, I did a local check out and it turns out the problem is trailing whitespace.

For fixing this kind of issue, I'd suggest to run the `glslangtests` binary directly with `--update-mode`, which will flush the expected result file with the actual output. Then with `git status`, we can see clearly what's changed and expected to be fixed.

(It's interesting that the standalone path ignores some whitespace characters. I'm not sure why that happens yet. Will dig out latter.)

File: glslang/MachineIndependent/linkValidate.cpp (hunk 1 above)
Comment (amdrexu) [inline glslang/MachineIndependent/linkValidate.cpp]: It is OK. I will correct this.

**Comment by amdrexu** [comment]
@antiagainst Thank you for this finding. I will use `glslangtests`.

**Comment by amdrexu** [comment]
@johnkslang I update the codes and tests.

**Comment by johnkslang** [comment]
> (It's interesting that the standalone path ignores some whitespace characters. I'm not sure why that happens yet. Will dig out later.)

The runtests diff uses options to ignore white space.

**Comment by antiagainst** [comment]
> The runtests diff uses options to ignore white space.

Ah, that's true and obvious via the `-b` command line option. Got it.

**Pull request merged**